### PR TITLE
Blog app - integration specs for views and fix n+1 problems

### PR DESCRIPTION
**Pull request by osmanwako**
# Inside this PR We did a system test and fix N+! problems
- We solved the N+1 problem when fetching all posts and their comments for a use
  - when the user fetches **user** index page before fixing N+! problems
userb1
  - when the user fetches **user** index page after fixing N+! problems
usera1
  -  when the user fetches **user** show page before fixing N+! problems
userb1
 - when the user fetches **user** show page after fixing N+! problems
userb2
  - when the user fetches **post** index page before fixing N+! problems
postb1
  - when the user fetches **post** index page after fixing N+! problems
posta1
  -  when the user fetches **post** show page before fixing N+! problems
postb2
  - when the user fetches **post** show page after fixing N+! problems
posta2

### System Test
- Two test files were created inside **test/system** (users_test and posts_test) directory
 - To run ** rails test:system**

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -15,12 +15,11 @@ def new
   end
```
Comment (t-yanick) [inline app/controllers/posts_controller.rb:16]: - [ ] Great job working on this milestone.

To solve the N+1 problem, you were to do so in the `posts#index` actions of the `posts` controller and you could use `@posts = @user.posts.includes(:comments)`

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -15,12 +15,11 @@ def new
...
```
Comment (t-yanick) [inline app/controllers/posts_controller.rb:17]: - [ ] In your `[blog_application](https://github.com/osmanwako/blog_application/tree/integration_specs)/[test](https://github.com/osmanwako/blog_application/tree/integration_specs/test)/[system](https://github.com/osmanwako/blog_application/tree/integration_specs/test/system)
/posts_test.rb` you are missing under the` Post Index Page` a test that redirects to the post's show page when you click on a post.

**Review by t-yanick — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED

Hi @osmanwako and @vanina123 , 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Screenshots of N+1 solution added to PR body
:heavy_check_mark: Descriptive Readme file
:heavy_check_mark: Use of correct gitflow

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ichala — APPROVED** [review]
Hi Student  ,
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:  
 
68747470733a2f2f6d65646961322e67697068792e636f6d2f6d656469612f6c3366516333614356304871707a5763552f67697068792e676966

# To Highlight 💪🏻

- Changes Implemented ✔️ 


 Good job implementing the requirements  :+1: Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Optional Suggestions  :

- N/A


**********
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag  ichala in your question so I can receive the notification.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**